backend/main-service/src/main/java/kkakka/mainservice/product/domain/repository/ProductRepositorySupport.java

```diff
@@ -40,6 +41,13 @@ public Page<Product> findBySearch(SearchWords searchWords, Pageable pageable) {
         return new PageImpl<>(result, pageable, result.size());
     }
 
+    public List<Product> findTop10ByKeyword(String keyword) {
+        return queryFactory.selectFrom(QProduct.product)
+            .where(QProduct.product.name.contains(keyword))
+            .limit(10).fetch();
+
+    }
```

Comment:
```suggestion
    public List<Product> findTop10ByKeyword(String keyword) {
        return queryFactory.selectFrom(QProduct.product)
            .where(QProduct.product.name.contains(keyword))
            .limit(10).fetch();
    }
```
혹시 이 `10`이라는 숫자를 바꿔서 사용하게 될 일은 없을까요?
메서드를 사용하는 쪽에서 수를 정해주면 5개, 20개로 기능을 확장할 수 있을 것 같아요.
만약 수가 바뀌게 된다면 지금은 그때마다 `findTop5ByKeyword`, `findTop20ByKeyword`를 만들어야겠죠..

Reply:
저도 그 점에 대해서 헷갈렸는데 실제 서비스는 es에서 조회해서 기본 10개를 제공해주고 있지만 테스트를 위해서 같은 역할을 하는 메서드를 만들다 보니 고정해서 만들게 된 것 같습니다.
나중에 경우가 발생하면 변경하는 쪽으로 생각하고 있었는데 확장성에 있어 필요하다고 생각이 들면 수정하도록 하겠습니다!